src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/MemoryPoolViewBufferScope.cs

```diff
@@ -60,7 +65,7 @@ public ViewBufferValue[] GetSegment()
 
             try
             {
-                segment = _viewBufferPool.Rent(SegmentSize);
+                segment = _viewBufferPool.Rent(Math.Max(size, MinimumSize));
```

Comment:
There's some non-zero likelihood that we will move this to a public place next week. I don't see any reason to take this out :see_no_evil: